src/Components/Web.JS/src/Platform/Circuits/CircuitStreamingInterop.ts

```diff
@@ -9,12 +9,21 @@ export function sendJSDataStream(connection: HubConnection, data: ArrayBufferVie
         let numChunksUntilNextAck = 5;
         let lastAckTime = new Date().valueOf();
         try {
+            const byteLength = data instanceof Blob ? data.size : data.byteLength;
             let position = 0;
             let chunkId = 0;
 
-            while (position < data.byteLength) {
-                const nextChunkSize = Math.min(chunkSize, data.byteLength - position);
-                const nextChunkData = new Uint8Array(data.buffer, data.byteOffset + position, nextChunkSize);
+            while (position < byteLength) {
+                const nextChunkSize = Math.min(chunkSize, byteLength - position);
+                let nextChunkData: ArrayBuffer;
+
+                if (data instanceof Blob) {
```

Comment:
nit: Instead of continuously checking this on every iteration, move this check outside of the loop and write two separate (different) loops based on whatever case you are doing

Reply:
c682d193ca